eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -671,6 +678,9 @@ def model(self, matches):
             elif key == "regulation":
                 if regulation:
                     raise UnspecifiedParsingError("Multiple regulations")
+                # check if regulation is in label
+                if not value.check_labels(self.labels):
+                    raise RegulationParsingError("Regulation label not found")
```

Comment:
Perhaps it's better to raise this Exception directly within individual `check_labels` functions? That way you can, for example, include what label was not found etc.